server/src/DBTablesMonitoring.cc

```diff
@@ -2101,9 +2099,15 @@ void DBTablesMonitoring::updateHosts(const HostInfoList &hostInfoList,
 	HostInfoListConstIterator newHostsItr = hostInfoList.begin();
 	for (; newHostsItr != hostInfoList.end(); ++newHostsItr) {
 		const HostInfo &newHostInfo = *newHostsItr;
-		if (currValidHosts.erase(newHostInfo.id) >= 1) {
-			// The host already exits. We have nrothing to do.
-			continue;
+		HostIdHostInfoMapIterator currHostItr = currValidHosts.find(newHostInfo.id);
+		if (currHostItr != currValidHosts.end()) {
+			const HostInfo &existHost = *currHostItr->second;
```

Comment:
'exist' is a verb. The combination 'exist' and 'host' is unnatural.
How about using 'existingHost'.
Or 'currHost' which is an opposite word of newHost(Info).